proto/Makefile

```diff
@@ -35,8 +35,8 @@ get-deps:
 
 build/proto.stamp: $(source_files)
 	protoc --version | tee /dev/stderr | \
-		perl -ne '(/libprotoc (?:3?\.)?(\d+)\.(?:\d+)/ && $$1>=15 && $$1<22) || \
-			(print("Please install protoc where version >= 3.15 and < 3.22\n") && exit 1)'
+		perl -ne '(/libprotoc (?:24\.)?(\d+)\.(\d+)/ && $$1>=24 && ($$1>24 || $$2>=3)) || \
```

Comment:
It looks like this regex change may not be doing what is intended. At some point protoc dropped the major version from the version number (altho `v3` tags are still being created). The `?:` part of the regex indicates a non-matching group, so the major part should not change it is just the minor and patch versions being assigned to $1 and $2. The max minor version check was dropped, does protobuf v26 also work here?

Reply:
More info on the versioning change:
https://protobuf.dev/news/2022-05-06/, basically the major version is reserved for language specific api updates, but previously protoc would report the major version as 3. It should probably stay as 3 or >=3.

We should also make sure whatever minor version is easily installable for devs. Right now we can `brew install protobuf@21`, and `brew install protobuf` currently installs v26.

Reply:
I'm thinking to leave it a >21 to keep it simple and the min version similar to ci?

Reply:
personally I would prefer stability to running the latest thing, unless there is a particular reason we need to upgrade.  eventually something will be mismatched between versions, causing breaking CI issues again.

Reply:
made sure we keep allowing > 3.24.x